spec/models/course_enrichment_spec.rb

```diff
@@ -91,36 +91,75 @@
   end
 
   describe 'validation for publish' do
-    subject { create(:course_enrichment, *course_enrichment_traits) }
+    let(:course_enrichment) do
+      create(:course_enrichment, :with_fee_based_course)
+    end
+    subject { course_enrichment }
 
     context 'fee based course' do
-      let(:course_enrichment_traits) { [:with_fee_based_course] }
-
       it { should_not validate_presence_of(:salary_details).on(:publish) }
       it { should validate_presence_of(:fee_uk_eu).on(:publish) }
-
-      context 'valid content' do
-        it { should be_valid :publish }
-      end
-
-      context 'invalid content' do
-        let(:course_enrichment_traits) { %i[with_fee_based_course with_invalid_content] }
-        it { should_not be_valid :publish }
+      it { should validate_presence_of(:about_course).on(:publish) }
+      it { should validate_presence_of(:qualifications).on(:publish) }
+
+
+      context 'invalid content exceed word count fields' do
+        let(:course_enrichment) do
+          create(:course_enrichment, :with_fee_based_course,
+                 about_course:  Faker::Lorem.sentence(400 + 1),
```

Comment:
I think this might be easier to follow if we did this per validation.

```
it 'validates maximum word count for about_course' do
  course_enrichment.about_course = Faker::Lorem.sentence(400 + 1)

  expect(course_enrichment).not_to be_valid
  expect(course_enrichment.errors[:about_course]).to be_present  
end
```

It's a bit more verbose but easier to follow. What do you think?

Reply:
You could check for the correct error message as well if you want.

Reply:
Also, then we don't need the `invalid_enrichment` trait.

Reply:
I did contemplate creating a custom validation matcher altogether based on https://github.com/thoughtbot/shoulda-matchers/blob/master/lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb probably over the top...

Reply:
That would be really nice actually :P but yeah extra work :)

Reply:
went for your original suggestion, custom validation matcher, was a bit over my head (a perhaps a pyjama challenge for you?)